frontend/packages/ux-editor/src/App.tsx

```diff
@@ -31,36 +35,29 @@ export function App() {
   const t = useText();
   const { org, app } = useParams();
   const [searchParams, setSearchParams] = useSearchParams();
-  const layoutPagesOrder = useSelector(
-    (state: IAppState) => state.formDesigner.layout.layoutSettings.pages.order
-  );
-  const layoutOrder = useSelector(
-    (state: IAppState) =>
-      state.formDesigner.layout.layouts[state.formDesigner.layout.selectedLayout]?.order
-  );
-
-  const selectedLayout = useSelector(
-    (state: IAppState) => state.formDesigner.layout.selectedLayout
-  );
-
-  const activeList = useSelector((state: IAppState) => state.formDesigner.layout.activeList);
 
   const datamodelQuery = useDatamodelQuery(org, app);
+  const formLayoutsQuery = useFormLayoutsQuery(org, app);
+  const formLayoutSettingsQuery = useFormLayoutSettingsQuery(org, app);
+  const addLayoutMutation = useAddLayoutMutation(org, app);
+
+  const selectedLayout = useSelector(selectedLayoutNameSelector);
+
+  const layoutPagesOrder = formLayoutSettingsQuery.data?.pages.order;
+  const layoutOrder = formLayoutsQuery.data?.[selectedLayout]?.order || {};
+
+  const activeList = useSelector((state: IAppState) => state.formDesigner.layout.activeList);
 
-  const isLayoutSettingsFetched = useSelector(
-    (state: IAppState) => state.formDesigner.layout.isLayoutSettingsFetched
-  );
-  const isLayoutFetched = useSelector((state: IAppState) => state.formDesigner.layout.fetched);
   const isWidgetFetched = useSelector((state: IAppState) => state.widgets.fetched);
 
   const dataModelFetchedError = datamodelQuery.error;
-  const layoutFetchedError = useSelector((state: IAppState) => state.formDesigner.layout.error);
+  const layoutFetchedError = formLayoutsQuery.error;
   const widgetFetchedError = useSelector((state: IAppState) => state.widgets.error);
 
   const componentIsReady =
-    isLayoutFetched &&
+    formLayoutsQuery.isSuccess &&
```

Comment:
Hvis vi destrukturerer objektet som nevnt i en tidligere kommentar, så trenger vi ikke å sjekke på isSuccess fordi "data/formLayouts" er undefined frem til det er success. :) 
```suggestion
    formLayouts &&
```